src/Symfony/Bundle/SecurityBundle/Command/UserPasswordEncoderCommand.php

```diff
@@ -86,13 +89,18 @@ protected function execute(InputInterface $input, OutputInterface $output)
         }
 
         if (!$salt) {
-            $saltQuestion = $this->createSaltQuestion($input, $output);
-            $salt = $helper->ask($input, $output, $saltQuestion);
+            if ($input->isInteractive()) {
+                $saltQuestion = $this->createSaltQuestion($input, $output);
+                $salt = $helper->ask($input, $output, $saltQuestion);
+            } else {
+                $salt = $this->generateSalt($output);
+            }
+
         }
 
         $output->writeln("\n <comment>Encoders are configured by user type in the security.yml file.</comment>");
 
-        if (!$userClass) {
+        if (!$userClass && $input->isInteractive()) {
             $userClassQuestion = $this->createUserClassQuestion($input, $output);
```

Comment:
No. The point was that this question isn't even required for using this command.
The fact the symfony's core user class is the default one is good, as, anyway, it was the one provided by default with the autocompleter before.
Now that the symfony's core user class is the default one, there is no issue while using the non-interactive command, and the interactive command get rid of a pretty useless question.

I think the whole condition and the `createUserClassQuestion` can be removed.

Sorry if I was unclear.